### DISCUSSION TITLE
Building menus and pages

### DISCUSSION THREAD

**Discussion by TangerineDreamer** [Bodmer/TFT_eSPI#932] [New processor/display support or capability request] [answered]
Hi,
I succeeded in building a 40 buttons menu,from the keypad sketch example,but i prefer to have a menu with
several pages,how can i do that?

**Comment by Bodmer** [comment, marked as answer]
There are no menu specific features in TFT_eSPI. I think you are looking for a GUI type display so perhaps try using [GUISlice,](https://github.com/ImpulseAdventure/GUIslice) or if you are experienced inmore advanced coding concepts try [lvgl](https://github.com/lvgl/lv_port_esp32). Both of these GUI libraries can be used with TFT_eSPI.  I do not provide help or support for them but I expect the authors will help.

**Comment by TangerineDreamer** [comment]
Hi,Bodmer,i find the keypad example rendering more exciting thant the other menu builders,i just wanted to
get some tips to divide all the 40 buttons in 10 button sets.
Anyway,i have another question,is it possible to load a bitmap,eg;a picture of several buttons,and then organize
some touch coordonates over it? In staying into tft_espi simple friendly coding :)

**Comment by Bodmer** [comment]
The touch functions have no knowledge of what is on the screen, so you can load an image, such as a jpeg, or draw the boxes to represent the buttons using the graphics functions. 

The screen touch is reported as an x,y coordinate, so you need to check this against the coordinate boundary box that the key is in. The button class does this for you. You can check in the sketch, see the On_Off_Button example.

**Comment by TangerineDreamer** [comment]
That's pretty cool :) ,it seems to be like the same stuff you can do with nextion
i did a collection of jauges and linear meters with nextion using bitmaps and the nextion jauge needle wich react
with analog input,do you think i can do the same,using the "needle" from the  linear meter example of tft espi 
and spin it over a bitmap of a meter?

**Comment by Bodmer** [comment]
Have you looked at the Animated Dial example? The requirement is that it must be possible to read the display RAM, see comments in that example.

See also the discussion and example code in #865
